### PR TITLE
[INLONG-3565][Manager] Unified the interface for flink plugin

### LINKED ISSUE #3565: [Improve] Unified call interface of Flink
### Description

Unified call interface of Flink ，unified  client interface and rest api  

### InLong Component

InLong Manager

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by LvJiancheng**
### Title Name: [INLONG-3565] [manager-plugins]unified the interface of flink

where *XYZ* should be replaced by the actual issue number.

Fixes #3565

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve.*

### unified the interface of flink

*Describe the modifications you've done.*

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

File: inlong-manager/manager-plugins/src/test/java/org/apache/inlong/manager/plugin/listener/SuspendSortListenerTest.java [hunk 1]
```diff
@@ -43,7 +43,7 @@ public void testListener() throws Exception {
 
         InlongGroupExtInfo inlongGroupExtInfo1 = new InlongGroupExtInfo();
         inlongGroupExtInfo1.setKeyName(InlongGroupSettings.SORT_URL);
-        inlongGroupExtInfo1.setKeyValue("127.0.0.1:8085");
+        inlongGroupExtInfo1.setKeyValue("9.135.80.112:8085");
```
Comment (gong) [inline inlong-manager/manager-plugins/src/test/java/org/apache/inlong/manager/plugin/listener/SuspendSortListenerTest.java:46]: modify `ip` to `127.0.0.1` for private

Comment (LvJiancheng) [reply inlong-manager/manager-plugins/src/test/java/org/apache/inlong/manager/plugin/listener/SuspendSortListenerTest.java]: done

File: inlong-manager/manager-plugins/src/main/java/org/apache/inlong/manager/plugin/flink/FlinkService.java [hunk 2]
```diff
@@ -177,7 +144,7 @@ public JobStatus getJobStatus(String jobId) {
             return status;
 
         } catch (Exception e) {
-            e.printStackTrace();
+            log.error("get job status error", e.getMessage());
```
Comment (healchow) [inline inlong-manager/manager-plugins/src/main/java/org/apache/inlong/manager/plugin/flink/FlinkService.java:147]: In the error log, we need to print all stack traces of exception, so it is recommended to use `log.error("get job status error: ", e);`.

Comment (LvJiancheng) [reply inlong-manager/manager-plugins/src/main/java/org/apache/inlong/manager/plugin/flink/FlinkService.java:147]: ok ,done

File: inlong-manager/manager-plugins/src/main/java/org/apache/inlong/manager/plugin/listener/SuspendSortListener.java [hunk 3]
```diff
@@ -59,7 +59,7 @@ public ListenerResult listen(WorkflowContext context) throws Exception {
                 InlongGroupExtInfo::getKeyName, InlongGroupExtInfo::getKeyValue));
         String sortExt = kvConf.get(InlongGroupSettings.SORT_PROPERTIES);
         if (StringUtils.isEmpty(sortExt)) {
-            String message = String.format("inlongGroupId:%s not add suspendProcess listener,sortProperties is empty",
+            String message = String.format("inlongGroupId:%s not add suspendProcess listener, sortProperties is empty",
```
Comment (healchow) [inline inlong-manager/manager-plugins/src/main/java/org/apache/inlong/manager/plugin/listener/SuspendSortListener.java:62]: Change the message to `String.format("groupId: %s not add suspendProcess listener, as the sortProperties is empty"`.

File: inlong-manager/manager-plugins/src/main/java/org/apache/inlong/manager/plugin/flink/FlinkService.java [hunk 4]
```diff
@@ -261,169 +228,45 @@ public String restore(FlinkInfo flinkInfo) {
...
-        } catch (Exception e) {
-            log.error("triggerSavepoint after stop job fail",e.getMessage());
-        }
-        return body;
-    }
+            RestClusterClient<StandaloneClusterId> client = getFlinkClient();
+            JobID jobID = JobID.fromHexString(jobId);
+            CompletableFuture<String> stopResult =
+                    client.stopWithSavepoint(jobID,requestBody.isDrain(),requestBody.getTargetDirectory());
+            result = stopResult.get();
```
[the remark this answers is not in the document]
Comment (LvJiancheng) [reply inlong-manager/manager-plugins/src/main/java/org/apache/inlong/manager/plugin/flink/FlinkService.java:249]: done

File: inlong-manager/manager-plugins/src/main/java/org/apache/inlong/manager/plugin/flink/FlinkService.java [hunk 5]
```diff
@@ -261,169 +228,45 @@ public String restore(FlinkInfo flinkInfo) {
...
-                .build();
+    public void cancelJobs(String jobId) {
         try {
-            client.newCall(request).execute();
-            log.info("the jobId:{} delete successfuly",jarId);
-        } catch (IOException e) {
-            log.warn("the jobId:{} delete successfuly",jarId);
+            RestClusterClient<StandaloneClusterId> client = getFlinkClient();
+            JobID jobID = JobID.fromHexString(jobId);
+            CompletableFuture<Acknowledge> result = client.cancel(jobID);
```
[the remark this answers is not in the document]
Comment (LvJiancheng) [reply inlong-manager/manager-plugins/src/main/java/org/apache/inlong/manager/plugin/flink/FlinkService.java]: ok

**Review by dockerzhang — APPROVED** [review]

File: inlong-manager/manager-plugins/src/main/java/org/apache/inlong/manager/plugin/flink/IntergrationTaskRunner.java [hunk 6]
```diff
@@ -126,7 +101,7 @@ public void run() {
                                 String jobId = flinkService.restore(flinkInfo);
                                 log.info("Restore job {} success in backend", jobId);
                             } catch (Exception e) {
-                                log.warn("Restore job failed in backend");
+                                log.error("Restore job failed in backend", e.getMessage());
```
Comment (healchow) [inline inlong-manager/manager-plugins/src/main/java/org/apache/inlong/manager/plugin/flink/IntergrationTaskRunner.java:104]: Should print stack traces of exception.

File: inlong-manager/manager-plugins/src/main/java/org/apache/inlong/manager/plugin/listener/RestartSortListener.java [hunk 7]
```diff
@@ -73,15 +73,15 @@ public ListenerResult listen(WorkflowContext context) throws Exception {
                 new TypeReference<Map<String, String>>(){});
         kvConf.putAll(result);
         if (StringUtils.isEmpty(kvConf.get(InlongGroupSettings.SORT_JOB_ID))) {
-            String message = String.format("inlongGroupId:%s not add restartProcess listener,SORT_JOB_ID is empty",
+            String message = String.format("inlongGroupId:%s not add restartProcess listener, SORT_JOB_ID is empty",
```
Comment (healchow) [inline inlong-manager/manager-plugins/src/main/java/org/apache/inlong/manager/plugin/listener/RestartSortListener.java:76]: Change to `"groupId [%s] not add restartProcess listener, as the sortJobId is empty"`.

File: inlong-manager/manager-plugins/src/main/java/org/apache/inlong/manager/plugin/listener/StartupSortListener.java [hunk 8]
```diff
@@ -64,7 +64,7 @@ public ListenerResult listen(WorkflowContext context) throws Exception {
         String dataFlows = kvConf.get(InlongGroupSettings.DATA_FLOW);
         String inlongGroupId = context.getProcessForm().getInlongGroupId();
         if (StringUtils.isEmpty(dataFlows)) {
-            String message = String.format("inlongGroupId:%s not exec startupProcess listener,dataflows is empty",
+            String message = String.format("inlongGroupId:%s not exec startupProcess listener, dataflows is empty",
```
Comment (healchow) [inline inlong-manager/manager-plugins/src/main/java/org/apache/inlong/manager/plugin/listener/StartupSortListener.java:67]: Change to `groupId [%s] not exec startupProcess listener, as the dataflows is empty`.

File: inlong-manager/manager-plugins/src/main/java/org/apache/inlong/manager/plugin/flink/FlinkService.java [hunk 9]
```diff
@@ -261,169 +226,43 @@ public String restore(FlinkInfo flinkInfo) {
...
+    public String stopJobs(String jobId, StopWithSavepointRequestBody requestBody) {
         try {
-            body = getReq(url);
-            log.info("triggerSavepoint success");
-            return body;
+            RestClusterClient<StandaloneClusterId> client = getFlinkClient();
+            JobID jobID = JobID.fromHexString(jobId);
+            CompletableFuture<String> stopResult =
+                    client.stopWithSavepoint(jobID,requestBody.isDrain(),requestBody.getTargetDirectory());
+            String result = stopResult.get();
```
Comment (healchow) [inline inlong-manager/manager-plugins/src/main/java/org/apache/inlong/manager/plugin/flink/FlinkService.java:246]: Just use `return stopResult.get();`.

File: inlong-manager/manager-plugins/src/main/java/org/apache/inlong/manager/plugin/flink/IntergrationTaskRunner.java (hunk 6 above)
Comment (LvJiancheng) [reply inlong-manager/manager-plugins/src/main/java/org/apache/inlong/manager/plugin/flink/IntergrationTaskRunner.java:104]: ok,done

File: inlong-manager/manager-plugins/src/main/java/org/apache/inlong/manager/plugin/listener/RestartSortListener.java (hunk 7 above)
Comment (LvJiancheng) [reply inlong-manager/manager-plugins/src/main/java/org/apache/inlong/manager/plugin/listener/RestartSortListener.java:76]: ok done

File: inlong-manager/manager-plugins/src/main/java/org/apache/inlong/manager/plugin/flink/FlinkService.java (hunk 9 above)
Comment (LvJiancheng) [reply inlong-manager/manager-plugins/src/main/java/org/apache/inlong/manager/plugin/flink/FlinkService.java:246]: ok,done

File: inlong-manager/manager-plugins/src/main/java/org/apache/inlong/manager/plugin/listener/StartupSortListener.java (hunk 8 above)
Comment (LvJiancheng) [reply inlong-manager/manager-plugins/src/main/java/org/apache/inlong/manager/plugin/listener/StartupSortListener.java:67]: done

**Review by healchow — APPROVED** [review]

**Pull request merged**